kratos/python_scripts/assign_vector_variable_to_entities_process.py

```diff
@@ -0,0 +1,120 @@
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return AssignVectorVariableToEntitiesProcess(Model, settings["Parameters"])
+
+## All the processes python should be derived from "Process"
+class AssignVectorVariableToEntitiesProcess(KratosMultiphysics.Process):
+    """This process assigns a given value (vector) to the entities belonging a certain submodelpart
+
+    Only the member variables listed below should be accessed directly.
+
+    Public member variables:
+    Model -- the container of the different model parts.
+    settings -- Kratos parameters containing solver settings.
+    """
+
+    def __init__(self, Model, settings ):
+        """ The default constructor of the class
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        Model -- the container of the different model parts.
+        settings -- Kratos parameters containing solver settings.
+        """
+
+        KratosMultiphysics.Process.__init__(self)
+
+        # The value can be a double or a string (function)
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "help"                 : "This process assigns a given value (vector) to the entities belonging a certain submodelpart",
+            "mesh_id"              : 0,
+            "model_part_name"      : "please_specify_model_part_name",
+            "variable_name"        : "SPECIFY_VARIABLE_NAME",
+            "interval"             : [0.0, 1e30],
+            "value"                : [10.0, "3*t", "x+y"],
+            "local_axes"           : {},
+            "entities"             : []
+        }
+        """
+        )
+
+        # Detect "End" as a tag and replace it by a large number
+        if settings.Has("interval"):
+            if settings["interval"][1].IsString():
+                if settings["interval"][1].GetString() == "End":
+                    settings["interval"][1].SetDouble(1e30) # = default_settings["interval"][1]
+                else:
+                    raise Exception("the second value of interval can be \"End\" or a number, interval currently:"+settings["interval"].PrettyPrintJsonString())
+
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        self.variable = KratosMultiphysics.KratosGlobals.GetVariable(settings["variable_name"].GetString())
+        if not isinstance(self.variable, KratosMultiphysics.Array1DVariable3) and not isinstance(self.variable, KratosMultiphysics.VectorVariable):
+            msg = "Error in AssignVectorToConditionProcess. Variable type of variable : " + settings["variable_name"].GetString() + " is incorrect . Must be a vector or array3"
+            raise Exception(msg)
+
+        self.model_part = Model[settings["model_part_name"].GetString()]
+
+        self.aux_processes = []
+
+        import assign_scalar_variable_to_entities_process
+
+        # Component X
+        if not settings["value"][0].IsNull():
```

Comment:
please use sth like `for i_dir in range(3):` for this and refactor, this would avoid lots of copy pasting, same in the other processes
=> next PR

Reply:
this is is almost every process